read SDRF file from a shell environment variable

pmb59 opened the pull request:
The method summariseAtlasExperiment in functions.R was modified to read SDRF file from a shell environment variable , if available

pmb59 committed:
included code to read SDRF file from a shell environment variable, if available

pinin4fjords commented on R/functions.R:
```diff
@@ -127,39 +127,47 @@ summarizeAtlasExperiment <- function( experimentAccession, atlasExperimentDirect
 	}
     
-    cat( paste( "Reading experiment config from", atlasExperimentXMLfile, "...\n" ) )
+  cat( paste( "Reading experiment config from", atlasExperimentXMLfile, "...\n" ) )
```
Could you fix the indentation levels please? The existing code is terrible for it as well, this is our chance to fix it! I mean throughout this routine, not just the highlighted line.

pmb59 replied:
yes, we noticed that yesterday

pinin4fjords commented on R/functions.R:
```diff
@@ -127,39 +127,47 @@ summarizeAtlasExperiment <- function( experimentAccession, atlasExperimentDirect
-	# Filename for SDRF.
-	sdrfBasename <- paste( experimentAccession, ".sdrf.txt", sep="" )
+  #read SDRF file from a shell environment variable if it exists (otherwise it returns "")
+  sdrfPath <- Sys.getenv(x = 'SDRF_PATH', unset = "", names = NA)
```
NA is probably a better solution than an empty string, then test with is.na() on the following line.

pcm32 replied:
We can drop the x here
```suggestion
  sdrfPath <- Sys.getenv('SDRF_PATH', unset = "", names = NA)
```

pinin4fjords commented on R/functions.R:
```diff
@@ -127,39 +127,47 @@ summarizeAtlasExperiment <- function( experimentAccession, atlasExperimentDirect
+  if (sdrfPath == ""){
+      
+    # Get the pipeline code from the experiment accession e.g. MTAB, MEXP
+	  pipeline <- gsub( "E-", "", experimentAccession )
+	  pipeline <- gsub( "-\\d+", "", pipeline )
```
```suggestion
	  pipeline <- unlist(strsplit(experimentAccession, '-'))[2]
```

pcm32 reviewed (commented):
LGTM in general. Could you add a small comment on the fairly non-compliant documentation that this function has and also on the documentation of summarizeAtlasExperiment, explaining which env var needs to be set and what is the behaviour with or without this env var being set. Thanks.

pmb59 committed:
- Indentation levels in functions.R fixed
- Drop the 'x' in Sys.getenv
- unset = NA in Sys.getenv
- L144-145 replaced by pipeline <- unlist(strsplit(experimentAccession, '-'))[2]

pmb59 committed:
Included a small comment on the function 'summarizeAtlasExperiment' to highlight that comments do not compliant R documentation format

pmb59 committed:
Modified the R Documentation file 'summarizeAtlasExperiment.Rd' to indicate that the shell environment variable SDRF_PATH can optionally be defined containing the path to the SDRF file

pcm32 commented on man/summarizeAtlasExperiment.Rd:
```diff
@@ -15,7 +15,8 @@
 	Path to the directory containing the Atlas XML config file and expressions
-	matrices
+	matrices. Optionally, a shell environment variable \code{SDRF_PATH} 
+	can be defined containing the path to the SDRF file
```
Great, this should do documentation wise.

pcm32 reviewed (approved):
On my side it looks good, and I see some indentation changes, but I haven't checked if this looks good on an IDE. Lets see what @pinin4fjords says.

pinin4fjords commented:
Indentation still all over the place- I'll fix it

pinin4fjords committed:
Fix indentation

pinin4fjords commented:
This file was full of mixed tabs and spaces for indentation which made quite illegible for me. This is due to the bad habits of one of our alumni, but @pmb59 make sure you've set your editor to use 4 spaces when you use the tab key.

pinin4fjords commented on R/functions.R:
```diff
@@ -94,129 +94,137 @@ createAtlasSiteConfig <- function( ) {
+    sdrfPath <- Sys.getenv( 'SDRF_PATH', unset = NA, names = NA )
+    
+    # if SDRF path not found
+    if (is.na(sdrfPath) == TRUE){
```
You don't need the == TRUE here

pinin4fjords committed:
Remove unnecessary TRUE

pinin4fjords committed:
Add missing full stop

pcm32 commented:
For the record, I have checked independently (mamba installed previous version + devtools + optparse, then used install_github from R on that conda env and run https://github.com/ebi-gene-expression-group/bulk-recalculations/blob/develop/bin/createAtlasExperimentSummary.R) and it is working well:

```
(test-sdrf-path) bash-4.4$ Rscript script.R --source E-MTAB-5577 --accession E-MTAB-5577 --output my.Rdata
Reading experiment config from E-MTAB-5577/E-MTAB-5577-configuration.xml ...
Finished reading experiment config
Reading SDRF from /myPath/EXPERIMENT/MTAB/E-MTAB-5577/E-MTAB-5577.sdrf.txt ...
Getting characteristic and factor column indices
Checking for technical replicates
Locating assay names...
Found assay names. Subsetting SDRF...
Finished subsetting SDRF.
No Unit columns found.
Fixing column headings...
Finished fixing column headings.
Removing duplicated columns...
Finished removing duplicated columns.
Applying new column headings...
Finished applying new column headings.
Removing duplicated rows...
Finished removing duplicated rows.
Finished reading SDRF
Creating experiment summary...
Experiment summary created.
(test-sdrf-path) bash-4.4$ env | grep 'SDRF_PATH'
SDRF_PATH=/myPath/EXPERIMENT/MTAB/E-MTAB-5577/E-MTAB-5577.sdrf.txt
```

with a copy of E-MTAB-5577 from analysis/differential/rna-seq.
